File: CivitasJava/src/civitas/SorpresaSalirCarcel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;
/**
 *
 * @author luis
 */
public class SorpresaSalirCarcel extends Sorpresa{
    
    private MazoSorpresas mazo;
    
    SorpresaSalirCarcel(MazoSorpresas mazo){
        super("SorpresaSalirCarcel");
        this.mazo = mazo;
    }
    
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual,todos)){
            informe(actual,todos);
            int contador = 0;
            for(int i = 0; i < todos.size(); i++){
                if(todos.get(i).tieneSalvoconducto()){
                    contador++;
                }
            }
            if(contador == 0){
                todos.get(actual).obtenerSalvoconducto(this);
                this.salirDelMazo();
            }
        }
    }
    
    void salirDelMazo(){
        mazo.inhabilitarCartaEspecial(this);
    }
    
    void usada(){
        mazo.habilitarCartaEspecial(this);
    }
    
    @Override
    public String toString(){
        String e = "\nNombre: SorpresaSalirCarcel";
        return e;
    }
}

File: CivitasRuby/lib/sorpresa_salir_carcel.rb
# encoding:utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class SorpresaSalirCarcel < Sorpresa

    def initialize(mazo)
      super("SorpresaSalirCarcel")
      @mazo = mazo
    end

    def aplicar_a_jugador(actual, todos)
      if jugador_correcto(actual, todos)
        informe(actual, todos)
        contador = 0
        for i in todos
          if i.tiene_salvo_conducto
            contador = contador + 1
          end
        end
        if contador == 0
          todos[actual].obtener_salvo_conducto(self)
          self.salir_del_mazo
        end
      end
    end

        
    def salir_del_mazo
        @mazo.inhabilitar_carta_especial(self)
    end
    
    def usada 
        @mazo.habilitar_carta_especial(self)
    end
    
    def to_s
      "\nNombre: #{@texto}"
    end
    
       
    public_class_method  :new
    
  end
end
